testcontainers-dapr/src/main/java/io/dapr/testcontainers/DaprContainer.java

```diff
@@ -0,0 +1,330 @@
+  }
+
+  public DaprContainer withAppChannelAddress(String appChannelAddress) {
+    this.appChannelAddress = appChannelAddress;
+    return this;
+  }
+
+  /**
+   * Get a map of Dapr component details.
+   * @param component A Dapr Component.
+   * @return Map of component details.
+   */
+  public Map<String, Object> componentToMap(Component component) {
+    Map<String, Object> componentProps = new HashMap<>();
+    componentProps.put("apiVersion", "dapr.io/v1alpha1");
+    componentProps.put("kind", "Component");
+
+    Map<String, String> componentMetadata = new LinkedHashMap<>();
+    componentMetadata.put("name", component.getName());
+    componentProps.put("metadata", componentMetadata);
+
+    Map<String, Object> componentSpec = new HashMap<>();
+    componentSpec.put("type", component.getType());
+    componentSpec.put("version", component.getVersion());
+
+    if (!component.getMetadata().isEmpty()) {
+      componentSpec.put("metadata", component.getMetadata());
+    }
+    componentProps.put("spec", componentSpec);
+    return componentProps;
+  }
+
+  /**
+   * Get a map of Dapr subscription details.
+   * @param subscription A Dapr Subscription.
+   * @return Map of subscription details.
+   */
+  public Map<String, Object> subscriptionToMap(Subscription subscription) {
+    Map<String, Object> subscriptionProps = new HashMap<>();
+    subscriptionProps.put("apiVersion", "dapr.io/v1alpha1");
+    subscriptionProps.put("kind", "Subscription");
+
+    Map<String, String> subscriptionMetadata = new LinkedHashMap<>();
+    subscriptionMetadata.put("name", subscription.getName());
+    subscriptionProps.put("metadata", subscriptionMetadata);
+
+    Map<String, Object> subscriptionSpec = new HashMap<>();
+    subscriptionSpec.put("pubsubname", subscription.getPubsubName());
+    subscriptionSpec.put("topic", subscription.getTopic());
+    subscriptionSpec.put("route", subscription.getRoute());
+
+    subscriptionProps.put("spec", subscriptionSpec);
+    return subscriptionProps;
+  }
+
+  @Override
+  protected void configure() {
+    super.configure();
+
+    if (getNetwork() == null) {
+      withNetwork(Network.newNetwork());
+    }
+    if (this.placementContainer == null) {
+      this.placementContainer = new DaprPlacementContainer(this.placementDockerImageName)
+          .withNetwork(getNetwork())
+          .withNetworkAliases(placementService)
+          .withReuse(this.shouldReusePlacement);
+      this.placementContainer.start();
+    }
+
+    withCommand(
+        "./daprd",
+        "-app-id",
+        appName,
+        "--dapr-listen-addresses=0.0.0.0",
+        "--app-protocol",
+        "http",
+        "-placement-host-address",
+        placementService + ":50006",
+        "--app-channel-address",
+        appChannelAddress,
+        "--app-port",
+        Integer.toString(appPort),
+        "--log-level",
+        daprLogLevel.toString(),
+        "-components-path",
+        "/components");
```

Comment:
Rename the folder also to "dapr-resources" or something.

Reply:
done